media_kit_video/lib/src/video_controller_android.dart

```diff
@@ -74,13 +116,21 @@ class VideoControllerAndroid extends VideoController {
       return _controllers[handle]!;
     }
 
+    // Enforce software rendering in emulators.
+    final bool isEmulator = await _channel.invokeMethod('Utils.IsEmulator');
```

Comment:
I use the same logic to force SW rendering for the iOS simulator, however the logic is in `VideoOutput.swift`, maybe I should move the logic to `video_controller_native.dart` when the platform is iOS, it would make the code more consistent.
What do you think?

Reply:
I think your implementation is good already (no need for change). I think there's no point in looping back to Dart for this.

It's just that most implementation of Android's video embedding is within Dart (instead of Java/Kotlin) & we need to query this from Dart. A dedicated specialized class was necessary `VideoControllerAndroid` requiring abstraction of `VideoController`, as we know.